src/rpc/server/basic.zig

```diff
@@ -106,26 +125,62 @@ pub fn acceptAndServeConnection(server_ctx: *server.Context) !void {
                 return;
             },
 
-            .genesis_file => {},
+            .genesis_file => {
+                logger.err()
+                    .field("conn", conn.address)
+                    .logf("Attempt to get our genesis file", .{});
+                try request.respond("Genesis file get is not yet implemented", .{
+                    .status = .service_unavailable,
+                    .keep_alive = false,
+                });
+                return;
+            },
 
             .not_found => {},
         },
         .POST => {
-            logger.err().logf("{} tried to invoke our RPC", .{conn_address});
-            try request.respond("RPCs are not yet implemented", .{
-                .status = .service_unavailable,
-                .keep_alive = false,
-            });
+            if (head_info.content_type != .@"application/json") {
+                std.debug.panic("TODO: handle bad or missing content type", .{});
+            }
+
+            // make the server handle the 100-continue in case there is one
+            const any_reader = request.reader() catch |err| switch (err) {
+                error.HttpExpectationFailed => return,
+                else => |e| return e,
+            };
+            const req_reader = sig.utils.io.narrowAnyReader(
+                any_reader,
+                std.http.Server.Request.ReadError,
+            );
+            _ = req_reader; // autofix
+
+            switch (head_info.transfer_encoding) {
+                .none => {
+                    const content_len = head_info.content_len orelse
+                        std.debug.panic("TODO: handle content_len xor transfer_encoding", .{});
```

Comment:
are these setup for another PR?

Reply:
Yeah, this is just to act as a skeleton; currently there are no tests or anything that would trigger this panic.
Would have been part of this stack, but I got side tracked with other details.